src/main/java/io/appium/java_client/remote/MobileOptions.java

```diff
@@ -0,0 +1,514 @@
+     */
+    public T setApp(URL url) {
+        return setApp(url.toString());
+    }
+
+    /**
+     * Get the app location.
+     *
+     * @return String representing app location
+     * @see MobileCapabilityType#APP
+     */
+    public String getApp() {
+        return (String) getCapability(MobileCapabilityType.APP);
+    }
+
+    /**
+     * Set the automation engine to use.
+     *
+     * @param name is the name of the automation engine
+     * @return this MobileOptions, for chaining.
+     * @see MobileCapabilityType#AUTOMATION_NAME
+     */
+    public T setAutomationName(String name) {
+        return amend(MobileCapabilityType.AUTOMATION_NAME, name);
+    }
+
+    /**
+     * Get the automation engine to use.
+     *
+     * @return String representing the name of the automation engine
+     * @see MobileCapabilityType#AUTOMATION_NAME
+     */
+    public String getAutomationName() {
+        return (String) getCapability(MobileCapabilityType.AUTOMATION_NAME);
+    }
+
+    /**
+     * Set the app to move directly into Webview context.
+     *
+     * @return this MobileOptions, for chaining.
+     * @see MobileCapabilityType#AUTO_WEBVIEW
+     */
+    public T setAutoWebview() {
+        return setAutoWebview(true);
+    }
+
+    /**
+     * Set whether the app moves directly into Webview context.
+     *
+     * @param bool is whether the app moves directly into Webview context.
+     * @return this MobileOptions, for chaining.
+     * @see MobileCapabilityType#AUTO_WEBVIEW
+     */
+    public T setAutoWebview(boolean bool) {
+        return amend(MobileCapabilityType.AUTO_WEBVIEW, bool);
+    }
+
+    /**
+     * Get whether the app moves directly into Webview context.
+     *
+     * @return true if app moves directly into Webview context.
+     * @see MobileCapabilityType#AUTO_WEBVIEW
+     */
+    public boolean doesAutoWebview() {
+        return (boolean) getCapability(MobileCapabilityType.AUTO_WEBVIEW);
+    }
+
+    /**
+     * Set the app to clear system files.
+     *
+     * @return this MobileOptions, for chaining.
+     * @see MobileCapabilityType#CLEAR_SYSTEM_FILES
+     */
+    public T setClearSystemFiles() {
+        return setClearSystemFiles(true);
+    }
+
+    /**
+     * Set whether the app clears system files.
+     *
+     * @param bool is whether the app clears system files.
+     * @return this MobileOptions, for chaining.
+     * @see MobileCapabilityType#CLEAR_SYSTEM_FILES
+     */
+    public T setClearSystemFiles(boolean bool) {
+        return amend(MobileCapabilityType.CLEAR_SYSTEM_FILES, bool);
+    }
+
+    /**
+     * Get whether the app clears system files.
+     *
+     * @return true if the app clears system files.
+     * @see MobileCapabilityType#CLEAR_SYSTEM_FILES
+     */
+    public boolean doesClearSystemFiles() {
+        return (boolean) getCapability(MobileCapabilityType.CLEAR_SYSTEM_FILES);
+    }
+
+    /**
+     * Set the name of the device.
```

Comment:
this cap, actually, is only used on Sauce and it does not affect the actual `real` device, which is used for testing. One must use the `udid` one for this purpose.

Reply:
It isn't just for Sauce, but I'll update this to `Set the kind of mobile device or emulator to use`

Reply:
4ae42ed81cfa222c8067702c53db70d73056d730